int32 right_value
int32 center_value
int32 left_value
